input/postgres/statements.go

```diff
@@ -253,38 +269,60 @@ func GetStatements(ctx context.Context, server *state.Server, logger *util.Logge
 		}
 
 		if showtext {
-			select {
-			// Since normalizing can take time, explicitly check for cancellations
-			case <-ctx.Done():
-				return nil, nil, nil, ctx.Err()
-			default:
-				fingerprintAndNormalize(key, receivedQuery.String, server, statements, statementTextsByFp)
+			queryKeys = append(queryKeys, key)
+			queryStats = append(queryStats, stats)
+			queryLength = append(queryLength, len(receivedQuery.String))
+			tmpFile.WriteString(receivedQuery.String)
+		} else {
+			if ignoreIOTiming(postgresVersion, "") {
+				stats.BlkReadTime = 0
+				stats.BlkWriteTime = 0
 			}
+			statementStats[key] = stats
 		}
-		if ignoreIOTiming(postgresVersion, receivedQuery) {
-			stats.BlkReadTime = 0
-			stats.BlkWriteTime = 0
-		}
-		statementStats[key] = stats
 	}
 
 	if err = rows.Err(); err != nil {
 		return nil, nil, nil, err
 	}
 
+	tmpFile.Seek(0, io.SeekStart)
+	for idx, length := range queryLength {
+		bytes := make([]byte, length)
+		_, err = io.ReadFull(tmpFile, bytes)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		query := string(bytes)
```

Comment:
We could technically avoid the memory copy here (see e.g. https://www.reddit.com/r/golang/comments/14xvgoj/converting_string_byte/ for a way of doing it), but not sure if its worth saving on that (maybe worth a benchmark, not necessarily right now)